Docs: Improve Device Agent Intro Documentation (#4718)
FlowFuse/flowfuse: docs/device-agent/quickstart.md

      - flowfuse
---

# Quick Start Guide: Device Agent

This guide will walk you through the process of adding a device to FlowFuse, connecting it to the platform, and deploying your Node-RED flows remotely. FlowFuse's Device Agent helps unlock the power of your devices by allowing you to manage and deploy Node-RED flows running on those devices securely and remotely.

![FlowFuse Device Agent Workflow](./images/device-agent-workflow.png){data-zoomable}
_Workflow to onboard new Edge Devices through the FlowFuse Platform and Device Agent_

## Setup & Installation

### Step 1: Install Device Agent (On Device)

> NodeJS of version 20 or later is recommended, through the device agent supports NodeJS v14 or later.

Reviewer: Do we need to be prepared for Device Agent v3 here?

For example. should we state something like

```suggestion
NodeJS version 18 is is supported but version 20 or later is recommended. However, if necessary, device agent version 2.8 does support NodeJS versions 14 and 16.
```

I wanted to explicitly call out v18 as the minimum (since that is what is in the package json in device agent v3.x)

> NodeJS version 18 is is supported but version 20 or later is recommended. However, if necessary, device agent version 2.8 does support NodeJS versions 14 and 16.

Reviewer: ```suggestion
NodeJS version 18 is supported but version 20 or later is recommended. However, if necessary, device agent version 2.8 does support NodeJS versions 14 and 16.
```

"is is supported " DOH!

#### Linux/MacOS

```bash
sudo npm install -g @flowfuse/device-agent
```

> #### Windows

Reviewer: The above linux instruction includes `sudo` which infers administrator access however the windows version does not make this clear.

I appreciate the full install instructions are linked below however a nod to administrative command prompt would be a helpful hint to avoid 1st try failures.

```suggestion
#### Windows (Elevated command prompt)
```

Author: Rather than putting this in the header, I'll put it in as a line of text

Reviewer: ```suggestion
#### Windows

Issue the below command in an elevated command prompt:
```

Does this help?

    _Screenshot of the dialog form to fill out when registering a Device_
    
5. Click **Add**
6. You will be presented with a **Device Configuration** that you will need in the next step. _(Do not close this window just yet.)_

    ![Screenshot of the dialog shown to a suer when a Device is registered](./images/config_yml2a.png){data-zoomable}
    _Screenshot of the dialog shown to a suer when a Device is registered_

### Step 3: Connect the Device (On Device)

1. Open a Terminal/Command Prompt on the Device.
2. Paste the command presented when you registered your Device, and run it.

    ![Example output in a Terminal showing a device agent successfully connecting to the FlowFuse Platform](./images/device_cli.png){data-zoomable}
    _Example output in a Terminal showing a device agent successfully connecting to the FlowFuse Platform_

### Step 4: Start the Device Agent (On Device)

1. Start the Device Agent using the command instructed by the device agent output e.g:

   ```bash
   flowfuse-device-agent
   ```

Your device is now connected to FlowFuse and ready to be used.

## Deploy Flows to Device

> There are two approaches to running flows on your Devices.

Reviewer: ```suggestion
There are two approaches to deploying flows to your Devices.
```

This section header is "Deploy Flows to Device" - the original text stated "running flows on your Devices" which when i read it, I was thinking we were referring to how the device-agent runs the node-red flows (like as a service or in docker)

      - flowfuse
---

# Quick Start Guide: Device Agent

This guide will walk you through the process of adding a device to FlowFuse, connecting it to the platform, and deploying your Node-RED flows remotely. FlowFuse's Device Agent helps unlock the power of your devices by allowing you to manage and deploy Node-RED flows running on those devices securely and remotely.

![FlowFuse Device Agent Workflow](./images/device-agent-workflow.png){data-zoomable}
_Workflow to onboard new Edge Devices through the FlowFuse Platform and Device Agent_

## Setup & Installation

### Step 1: Install Device Agent (On Device)

NodeJS version 18 is supported but version 20 or later is recommended. However, if necessary, device agent version 2.8 does support NodeJS versions 14 and 16.

#### Linux/MacOS

```bash
sudo npm install -g @flowfuse/device-agent
```

#### Windows

Issue the below command in an elevated command prompt:

```bash
npm install -g @flowfuse/device-agent
```

> For full install instructions, follow the [install instructions here](/docs/device-agent/install.md).

Reviewer: should this line not be in the parent section?

or at least repeated on the linux subsection?

currently, it is nested under the windows section.
